### ISSUE TITLE
Asserts for message box answer combinations are too restrictive

### ISSUE THREAD

**Issue by Elmi77** [wxWidgets/wxWidgets#25010]
Right at the moment, when opening a message box, the combination of styles that can be used for confirming this message box are restricted to several variants that belong together:

so wxMessageDialogBase::SetMessageDialogStyle() has a bunch of asserts that try to exclude combinations that do not make sense. However, the assert

wxASSERT_MSG( ((style & wxYES_NO) == wxYES_NO) || !(style & wxYES_NO),
                      "wxYES and wxNO may only be used together" );

is too restrictive in my opinion. Imagine a message box

"Your configuration for blabla overlaps the settings for blablubb. This can cause an unexpected behaviour. When this is not done by intention, choose an other one for proper operation."

Here the combination of answer buttons "OK" and "No" do make sense:

OK - it was not done by intention and it will be changed
No - I will not change this as it is exactly what I wanted to do.

So to sum it up: in my opinion "wxOK" is quite similar to "wxYES" and therefore should be allowed together with "wxNO".

**Comment by imciner2** [comment]
In the mean time, you can always change the label on the `yes` button with the function [`SetYesNoLabels()`](https://docs.wxwidgets.org/3.2.5/classwx_message_dialog.html#aa23b25ebf18bf0915d438cc13b55cb23), although it would still return `wxID_YES` as the selected option if they click that button.
